azure/ConsiderationsForServiceDesign.md

```diff
@@ -0,0 +1,99 @@
+> 
+> :no_entry: **DO NOT** proactively add APIs for speculative features customers might want 
+
+> :white_check_mark: **DO** consider how your abstractions will be represented in different high-level languages.
+
+> :white_check_mark: **DO** develop code examples in at least one dynamically typed language (for example, Python or JavaScript) and one statically typed language (for example, Java or C#) to illustrate your abstractions and high-level language representations.
+
+### Start with your API Definition
+Understanding how your service is used and defining its model and interaction patterns--its API--should be one of the earliest activities a service team undertakes. It reflects the abstractions & naming decisions and makes it easy for developers to implement the hero scenarios.  
+
+> :white_check_mark: **DO** provide an [OpenAPI Definition](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/2.0.md) (with [autorest extensions](https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md)) describing the service. The OpenAPI definition is a key element of the Azure SDK plan and is essential for documentation, usability and discoverability of services.
+
+### Use Previews to Iterate 
+ Before releasing your API plan to invest significant design effort, get customer feedback, & iterate through multiple preview releases. This is especially important for V1 as it establishes the abstractions and patterns that developers will use to interact with your service.
+
+> :ballot_box_with_check: **YOU SHOULD**  write and test hypotheses about how your customers will use the API.
+> 
+> :ballot_box_with_check: **YOU SHOULD**  release and evaluate a minimum of 2 preview versions prior to the first GA release.
+> 
+> :ballot_box_with_check: **YOU SHOULD**  identify key scenarios or design decisions in your API that you want to test with customers, and ask customers for feedback and to share relevant code samples.
+> 
+> :ballot_box_with_check: **YOU SHOULD**  consider doing a *code with* exercise in which you actively develop with the customer, observing and learning from their API usage.
+> 
+> :ballot_box_with_check: **YOU SHOULD**  capture what you have learned during the preview stage and share these findings with your team and with the API Stewardship Board.
+
+### Avoid Surprises
+A major inhibitor to adoption and usage is when an API behaves in an unexpected way. Often, these are subtle design decisions that seem benign at the time, but end up introducing significant downstream friction for developers.
+
+> :ballot_box_with_check: **YOU SHOULD** avoid polymorphism, especially in the response. An endpoint __SHOULD__ work with a single type to avoid problems during SDK creation.
+>
+> :ballot_box_with_check: **YOU SHOULD** make [Collections](./Guidelines.md#collections) easy to work with. Collections are a common source of review comments. It is important to handle them in a consistent manner within your service.
```

Comment:
This link is very good, but I was surprised (no pun intended) that the link too me to the collection guidelines; I thought it would be just a definition of what a collection is.

Reply:
```suggestion
> :ballot_box_with_check: **YOU SHOULD** make collections easy to work with. Collections are a common source of review comments. It is important to handle them in a consistent manner within your service. See [Collections](./Guidelines.md#collections). 
```